### DISCUSSION TITLE
Copilot AI auto completing sentences with quite limit political opinions

### DISCUSSION THREAD

**Discussion by leodumont444** [community/community#62296] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hey, we were having fun with Copilot this afternoon, a friend that isn't code friendly wrote things on a projet, as a commentary.
He started to write some "funny" sentences in french, to experience. He wrote (in french) "i'll vote" and copilot completed with "Lepen in 2027", Lepen being a real candidate to the 2027 french president elections. It made an other friend laugh so he tried with something rac_st, starting with "arabs are" and copilot ended the sentence with "thieves", which is kinda racist in my opinion, he also wrote "h_tler was" and copilot completed with "a good guy". 

Kinda scary to see that an AI used worldwide to CODE is having limit political opinions and is near racist.

**Comment by Akash1134** [comment]
Hi @leodumont444 ,

Public code may contain insecure coding patterns, bugs, or references to outdated APIs or idioms. When GitHub Copilot synthesizes code suggestions based on this data, it can also synthesize code that contains these undesirable patterns. This is something we care a lot about at GitHub, and in recent years we’ve provided tools such as GitHub Actions, Dependabot, and CodeQL to open source projects to help improve code quality. On top of that, you mentioned Copilot was not used for coding in this scenario.

Of course, you should always use GitHub Copilot together with good testing and code review practices and security tools, as well as your own judgment.

Please report incidents (e.g., offensive output, code vulnerabilities, apparent personal information in code generation) directly to copilot-safety@github.com so that we can improve our safeguards. GitHub takes safety and security very seriously and we are committed to continually improving.
